src/seedsigner/helpers/kef.py

```diff
@@ -0,0 +1,565 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from Cryptodome.Cipher import AES
+import hashlib
+
+
+# KEF: AES, MODEs VERSIONS, MODE_NUMBERS, and MODE_IVS are defined here
+#  to disable a MODE: set its value to None
+#  to disable a VERSION: set its value to None
+MODE_ECB = AES.MODE_ECB
+MODE_CBC = AES.MODE_CBC
+MODE_CTR = AES.MODE_CTR
+MODE_GCM = AES.MODE_GCM
+VERSIONS = {
+    # initial versions: released 2023.08 to encrypt bip39 entropy bytes
+    0: {
+        "name": "AES-ECB v1",
+        "mode": MODE_ECB,
+        "auth": -16,
+    },
+    1: {
+        "name": "AES-CBC v1",
+        "mode": MODE_CBC,
+        "auth": -16,
+    },
+    # AES in ECB mode
+    5: {
+        # smallest ECB ciphertext, w/ unsafe padding: for high entropy mnemonics, passphrases, etc
+        "name": "AES-ECB",
+        "mode": MODE_ECB,
+        "auth": 3,
+    },
+    6: {
+        # safe padding: for mid-sized plaintext w/o duplicate blocks
+        "name": "AES-ECB +p",
+        "mode": MODE_ECB,
+        "pkcs_pad": True,
+        "auth": -4,
+    },
+    7: {
+        # compressed, w/ safe padding: for larger plaintext; may compact otherwise duplicate blocks
+        "name": "AES-ECB +c",
+        "mode": MODE_ECB,
+        "pkcs_pad": True,
+        "auth": -4,
+        "compress": True,
+    },
+    # AES in CBC mode
+    10: {
+        # smallest CBC cipherext, w/ unsafe padding: for mnemonics, passphrases, etc
+        "name": "AES-CBC",
+        "mode": MODE_CBC,
+        "auth": 4,
+    },
+    11: {
+        # safe padding: for mid-sized plaintext
+        "name": "AES-CBC +p",
+        "mode": MODE_CBC,
+        "pkcs_pad": True,
+        "auth": -4,
+    },
+    12: {
+        # compressed, w/ safe padding: for larger plaintext
+        "name": "AES-CBC +c",
+        "mode": MODE_CBC,
+        "pkcs_pad": True,
+        "auth": -4,
+        "compress": True,
+    },
+    # AES in CTR stream mode
+    15: {
+        # doesn't require padding: for small and mid-sized plaintext
+        "name": "AES-CTR",
+        "mode": MODE_CTR,
+        "pkcs_pad": None,
+        "auth": -4,
+    },
+    16: {
+        # compressed: for larger plaintext
+        "name": "AES-CTR +c",
+        "mode": MODE_CTR,
+        "pkcs_pad": None,
+        "auth": -4,
+        "compress": True,
+    },
+    # AES in GCM stream mode
+    20: {
+        # doesn't require padding: for small and mid-sized plaintext
+        "name": "AES-GCM",
+        "mode": MODE_GCM,
+        "pkcs_pad": None,
+        "auth": 4,
+    },
+    21: {
+        # compressed: for larger plaintext
+        "name": "AES-GCM +c",
+        "mode": MODE_GCM,
+        "pkcs_pad": None,
+        "auth": 4,
+        "compress": True,
+    },
+}
+MODE_NUMBERS = {
+    "AES-ECB": MODE_ECB,
+    "AES-CBC": MODE_CBC,
+    "AES-CTR": MODE_CTR,
+    "AES-GCM": MODE_GCM,
+    "AES-ECB v1": MODE_ECB,
+    "AES-CBC v1": MODE_CBC,
```

Comment:
I'm not certain that the 2 lines above are necessary, then again I'm not sure how they are used in the seedsigner fork.   In krux, MODE_NUMBERS are necessary to keep track of user settings preferences for which mode the user prefers... so that it can be mapped to the mode stored in the VERSIONS dict.